### DISCUSSION TITLE
VOD not playing after s3 recording enabled in Ant Media server

### DISCUSSION THREAD

**Discussion by yashtandon113** [ant-media/Ant-Media-Server#4876] [FAQs] [answered]
I have enabled s3 recording in application settings on dashboard but when trying to play VOD from dashboard it gives error "The media could not be loaded, either because the server or network failed or because the format is not supported."

![image](https://user-images.githubusercontent.com/82374739/223516721-8deffb9c-9fa4-4572-8cfa-10f84269c662.png)

Also it is not working with iframe/Embedded code.

**Comment by Mohit-3196** [comment, marked as answer]
This error _The media could not be loaded, either because the server or network failed or because the format is not supported_ generally comes when **HTTP forwarding** is not configured on Ant Media Server for S3 Recording.

**HTTP forwarding** is implemented to forward incoming HTTP requests to any other place. It's generally used for forwarding the incoming request to storage like S3.

Solution:
Here's how HTTP forwarding can be configured on Ant Media Server:

Open the file _{AMS-DIR} / webapps / {APPLICATION} / WEB-INF / red5-web.properties_ with your text editor.
Add `settings.httpforwarding.extension=mp4` to the file.
Add the base URL with `settings.httpforwarding.baseURL=https://{YOUR_DOMAIN}` for forwarding.

Please replace _{YOUR_DOMAIN}_ with your own URL. Please pay attention that there is no leading or trailing white spaces.
e.g.,
If you are using **AWS S3 bucket**, {YOUR_DOMAIN} will be like:
_{s3BucketName}.s3.{awsLocation}.amazonaws.com_

If you are using **Digital Ocean Spaces**, {YOUR_DOMAIN} will be like:
_{BucketName}.{BucketLocation}.digitaloceanspaces.com_

Save the file and
restart the Ant Media Server with `sudo service antmedia restart`

If it's configured properly, your incoming MP4 requests such as _https://{SERVER_DOMAIN}:5443/{APPLICATION_NAME}/streams/vod.mp4_ will be forwarded to _https://{YOUR_DOMAIN_HERE}/streams/vod.mp4_

**CORS parameters** of AWS S3 bucket should be modified so that the requests that are coming from another origin after **embedding streams with iFrame** to play the VODs can be processed.

Go to your _AWS -> Services -> S3 -> Buckets -> "Your Bucket" -> Permissions ->_ And at the bottom of the page there is _Cross-origin resource sharing (CORS)._
_Click Edit->_ and paste the code provided below:

```
[
    {
        "AllowedHeaders": [
            "*"
        ],
        "AllowedMethods": [
            "HEAD",
            "GET",
            "PUT",
            "POST",
            "DELETE"
        ],
        "AllowedOrigins": [
            "*"
        ],
        "ExposeHeaders": []
    }
]
```

For more detailed information regarding S3 uploads, HTTP forwarding and other related topics checkout this document https://antmedia.io/docs/guides/recording-live-streams/s3-integration-http-forwarding/#http-forwarding
